Microsoft.Research/Contracts/MsCorlib/System.Threading.Tasks.Task.cs

```diff
@@ -1545,7 +1574,162 @@ public static Task<TResult> Run<TResult>(Func<TResult> function, CancellationTok
       return null;
 
     }
-    
+    //
+    // Summary:
+    //     Creates a System.Threading.Tasks.Task`1 that's completed successfully with the
+    //     specified result.
+    //
+    // Parameters:
+    //   result:
+    //     The result to store into the completed task.
+    //
+    // Type parameters:
+    //   TResult:
+    //     The type of the result returned by the task.
+    //
+    // Returns:
+    //     The successfully completed task.
+    [Pure]
+    public static Task<TResult> FromResult<TResult>(TResult result)
+    {
+      Contract.Ensures(Contract.Result<Task<TResult>>() != null);
+      Contract.Ensures(Contract.Result<Task<TResult>>().Exception == null);
+      Contract.Ensures(!Contract.Result<Task<TResult>>().IsCanceled);
+      Contract.Ensures(Contract.Result<Task<TResult>>().IsCompleted);
+      Contract.Ensures(!Contract.Result<Task<TResult>>().IsFaulted);
+      Contract.Ensures(Contract.Result<Task<TResult>>().Status == TaskStatus.RanToCompletion);
+
+      return default(Task<TResult>);
+    }
+#endif
+
+#if NETFRAMEWORK_4_6
+    //
+    // Summary:
+    //     Creates a System.Threading.Tasks.Task that's completed due to cancellation with
+    //     a specified cancellation token.
+    //
+    // Parameters:
+    //   cancellationToken:
+    //     The cancellation token with which to complete the task.
+    //
+    // Returns:
+    //     The canceled task.
+    [Pure]
+    public static Task FromCanceled(CancellationToken cancellationToken)
+    {
+      Contract.Requires(cancellationToken.IsCancellationRequested);
+      Contract.Ensures(Contract.Result<Task>() != null);
+      Contract.Ensures(Contract.Result<Task>().Exception == null);
+      Contract.Ensures(Contract.Result<Task>().IsCanceled);
+      Contract.Ensures(Contract.Result<Task>().IsCompleted);
+      Contract.Ensures(!Contract.Result<Task>().IsFaulted);
+      Contract.Ensures(Contract.Result<Task>().Status == TaskStatus.Canceled);
+      Contract.EnsuresOnThrow<ArgumentOutOfRangeException>(true, "Cancellation has not been requested for cancellationToken; its IsCancellationRequested property is false.");
```

Comment:
What is the purpose of this kinds of contracts? It never fails at runtime and it seems that it has no value for static analyzer. Is it some kind of documentation?